src/main/java/org/sagebionetworks/bridge/services/ParticipantFileService.java

```diff
@@ -118,6 +118,10 @@ public ParticipantFile createParticipantFile(String appId, String userId, Partic
 
         participantFileDao.uploadParticipantFile(file);
 
+        // Deleting any previous object prevents a user from updating the ParticipantFile
+        // but leaving the previous object on S3.
+        s3Client.deleteObject(bucketName, getFilePath(file));
```

Comment:
It's probably worth adding a `verify(mockS3Client).deleteObject()` in the unit tests.

Also, what happens if the object doesn't exist in S3? Does it do nothing or does it throw?

Reply:
That's good call, I'll update the unit tests.

If the object doesn't exist in S3, it just does nothing. The delete route uses that method as well and has a comment saying it returns as successful even if it didn't find anything to delete.

Reply:
Updated to verify that deleteObject was run through the mockS3Client.